File: mvpreaderjava/src/main/java/com/example/mvpreaderjava/di/component/AppComponent.java
package com.example.mvpreaderjava.di.component;

import com.example.mvpreaderjava.MyApplication;
import com.example.mvpreaderjava.di.module.AppModule;
import com.example.mvpreaderjava.di.module.HttpModule;
import com.example.mvpreaderjava.modle.DataManager;
import com.example.mvpreaderjava.modle.helper.HttpManager;

import javax.inject.Singleton;

import dagger.Component;

/**
 * Created by sw116 on 2018/4/19.
 */
@Component(modules = {HttpModule.class, AppModule.class})
@Singleton
public interface AppComponent {

    MyApplication getApp();

    DataManager getDataManager(); //数据中心

    HttpManager getHttpManager();//api接口

}

File: app/src/main/java/com/savion/mvpreader/di/component/AppComponent.kt
package com.savion.mvpreader.di.component

import android.app.Application
import com.savion.mvpreader.di.module.AppModule
import com.savion.mvpreader.di.module.HttpModule
import com.savion.mvpreader.model.http.service.HttpManager
import dagger.Component
import javax.inject.Singleton

/**
 * Created by savion on 2018/4/10.
 */
@Component(modules = arrayOf(AppModule::class, HttpModule::class))
@Singleton
interface AppComponent {

    fun getApp(): Application

    fun getHttpManager(): HttpManager

}